sw/device/silicon_creator/lib/manifest.S

```diff
@@ -0,0 +1,17 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+/**
+ * Manifest for boot stages stored in flash.
+ */
+
+  #include "sw/device/silicon_creator/lib/manifest_size.h"
+
+  .section .manifest, "a"
+  .globl kManifest
+
+kManifest:
+  .rept MANIFEST_SIZE
+  .byte 0x0
+  .endr
```

Comment:
So the problem with this is that this generates a TU that LLVM could potentially look at, and perhaps optimize all manifest acceses out to zero. What I think we actually want is something at the linker level like so:
```
SECTION {
  .manifest 0xwhatever: {
    kManifest = .;
    . += MANIFEST_SIZE;
  }
}
```

Unfortunately we'd need to define `MANFIEST_SIZE` twice, which is... unfortunate, to say the least. Maybe we can leave this like this for now and file a bug...

Reply:
Thanks for pointing this out. I think this can cause an issue only if boot stages access their own manifests, which they don't right now. Created #6814.